Release 1.5.0 to develop (#1195)
ginkgo-project/ginkgo: CHANGELOG.md

git log --first-parent
```

## Version 1.5.0

The Ginkgo team is proud to announce the new Ginkgo minor release 1.5.0. This release brings many important new features such as:
- MPI-based multi-node support for all matrix formats and most solvers;
- full DPC++/SYCL support,
- functionality and interface for GPU-resident sparse direct solvers,

> - an inerface for wrapping solvers with automatic scaling and reordering applied,

Reviewer: ```suggestion
- an interface for wrapping solvers with scaling and reordering applied,
```

Reviewer: I'm not sure automatic is the right adjective to describe this behavior, since it needs to be called explicitly.

> - a new algebraic Multigrid solver,

Reviewer: ```suggestion
- a new algebraic Multigrid solver/preconditioner,
```

+ Windows
  + MinGW and Cygwin: GCC 5.5+
  + Microsoft Visual Studio: VS 2019
  + CUDA module: CUDA 9.2+, Microsoft Visual Studio
  + OpenMP module: MinGW or Cygwin.

Algorithm and important feature additions:
+ Add MPI-based multi-node support, with support for all matrix formats and solvers (except GMRES and IDR). ([#676](https://github.com/ginkgo-project/ginkgo/pull/676), [#908](https://github.com/ginkgo-project/ginkgo/pull/908), [#909](https://github.com/ginkgo-project/ginkgo/pull/909), [#932](https://github.com/ginkgo-project/ginkgo/pull/932), [#951](https://github.com/ginkgo-project/ginkgo/pull/951), [#961](https://github.com/ginkgo-project/ginkgo/pull/961), [#971](https://github.com/ginkgo-project/ginkgo/pull/971), [#976](https://github.com/ginkgo-project/ginkgo/pull/976), [#985](https://github.com/ginkgo-project/ginkgo/pull/985), [#1007](https://github.com/ginkgo-project/ginkgo/pull/1007), [#1030](https://github.com/ginkgo-project/ginkgo/pull/1030), [#1054](https://github.com/ginkgo-project/ginkgo/pull/1054), [#1100](https://github.com/ginkgo-project/ginkgo/pull/1100), [#1148](https://github.com/ginkgo-project/ginkgo/pull/1148))

> + Porting the remaining algorithms (preconditioners like ISAI, Jacobi, Multigrid, ParILU(T) and ParIC(T)) to DPC++/SYCL, update to SYCL 2020, and improve support and performance ([#896](https://github.com/ginkgo-project/ginkgo/pull/896), [#924](https://github.com/ginkgo-project/ginkgo/pull/924), [#928](https://github.com/ginkgo-project/ginkgo/pull/928), [#929](https://github.com/ginkgo-project/ginkgo/pull/929), [#933](https://github.com/ginkgo-project/ginkgo/pull/933), [#943](https://github.com/ginkgo-project/ginkgo/pull/943), [#960](https://github.com/ginkgo-project/ginkgo/pull/960), [#1057](https://github.com/ginkgo-project/ginkgo/pull/1057), [#1110](https://github.com/ginkgo-project/ginkgo/pull/1110),  [#1142](https://github.com/ginkgo-project/ginkgo/pull/1142)).

Reviewer 2: ```suggestion
+ Porting the remaining algorithms (preconditioners like ISAI, Jacobi, Multigrid, ParILU(T) and ParIC(T)) to DPC++/SYCL, update to SYCL 2020, and improve support and performance ([#896](https://github.com/ginkgo-project/ginkgo/pull/896), [#924](https://github.com/ginkgo-project/ginkgo/pull/924), [#928](https://github.com/ginkgo-project/ginkgo/pull/928), [#929](https://github.com/ginkgo-project/ginkgo/pull/929), [#933](https://github.com/ginkgo-project/ginkgo/pull/933), [#943](https://github.com/ginkgo-project/ginkgo/pull/943), [#960](https://github.com/ginkgo-project/ginkgo/pull/960), [#1057](https://github.com/ginkgo-project/ginkgo/pull/1057), [#1110](https://github.com/ginkgo-project/ginkgo/pull/1110),  [#1142](https://github.com/ginkgo-project/ginkgo/pull/1142))
```

+ Add a logger highlighting inefficient allocation and copy patterns ([#1035](https://github.com/ginkgo-project/ginkgo/pull/1035))
+ Improved and optimized test random matrix generation ([#954](https://github.com/ginkgo-project/ginkgo/pull/954), [#1032](https://github.com/ginkgo-project/ginkgo/pull/1032))
+ Better CSR strategy defaults ([#969](https://github.com/ginkgo-project/ginkgo/pull/969))
+ Add `move_from` to `PolymorphicObject` ([#997](https://github.com/ginkgo-project/ginkgo/pull/997))
+ Remove unnecessary device_guard usage ([#956](https://github.com/ginkgo-project/ginkgo/pull/956))
+ Improvements to the generic accessor for mixed-precision ([#727](https://github.com/ginkgo-project/ginkgo/pull/727))
+ Add a naive lower triangular solver implementation for CUDA ([#764](https://github.com/ginkgo-project/ginkgo/pull/764))
+ Add support for int64 indices from CUDA 11 onward with SpMV and SpGEMM ([#897](https://github.com/ginkgo-project/ginkgo/pull/897))
+ Add an L1 norm implementation ([#900](https://github.com/ginkgo-project/ginkgo/pull/900))
+ Add reduce_add for arrays ([#831](https://github.com/ginkgo-project/ginkgo/pull/831))

> + Utility to simplify Dense View creation from an existing Dense vector ([#1136](https://github.com/ginkgo-project/ginkgo/pull/1136)).

Reviewer 3: ```suggestion
+ Add utility to simplify Dense View creation from an existing Dense vector ([#1136](https://github.com/ginkgo-project/ginkgo/pull/1136)).
```

> + Adds a custom transpose implementation for Fbcsr and Csr transpose for vendor-unsupported types. ([#1123](https://github.com/ginkgo-project/ginkgo/pull/1123))

Reviewer 3: ```suggestion
+ Add a custom transpose implementation for Fbcsr and Csr transpose for unsupported vendor types. ([#1123](https://github.com/ginkgo-project/ginkgo/pull/1123))
```

+ Deterministic IDR random init ([#1116](https://github.com/ginkgo-project/ginkgo/pull/1116))

> + Moves the algorithm choice for triangular solvers from Csr::strategy_type to a factory parameter. ([#1088](https://github.com/ginkgo-project/ginkgo/pull/1088))

Reviewer 3: ```suggestion
+ Move the algorithm choice for triangular solvers from Csr::strategy_type to a factory parameter. ([#1088](https://github.com/ginkgo-project/ginkgo/pull/1088))
```

+ Update CUDA archCoresPerSM ([#1175](https://github.com/ginkgo-project/ginkgo/pull/1116))
+ Add kernels for Csr sparsity pattern lookup ([#994](https://github.com/ginkgo-project/ginkgo/pull/994))
+ Differentiate between structural and numerical zeros in Ell/Sellp ([#1027](https://github.com/ginkgo-project/ginkgo/pull/1027))

> + Adds a binary IO format for matrix data ([#984](https://github.com/ginkgo-project/ginkgo/pull/984))

Reviewer 3: ```suggestion
+ Add a binary IO format for matrix data ([#984](https://github.com/ginkgo-project/ginkgo/pull/984))
```

+ Add a tuple zip_iterator implementation ([#966](https://github.com/ginkgo-project/ginkgo/pull/966))
+ Simplify kernel stubs and declarations ([#888](https://github.com/ginkgo-project/ginkgo/pull/888))
+ Simplify GKO_REGISTER_OPERATION with lambdas ([#859](https://github.com/ginkgo-project/ginkgo/pull/859))
+ Simplify copy to device in tests and examples ([#863](https://github.com/ginkgo-project/ginkgo/pull/863))
+ More verbose output to array assertions ([#858](https://github.com/ginkgo-project/ginkgo/pull/858))

> + Parallel Jacobi compilation ([#871](https://github.com/ginkgo-project/ginkgo/pull/871))

Reviewer 3: ```suggestion
+ Allow parallel compilation for Jacobi kernels ([#871](https://github.com/ginkgo-project/ginkgo/pull/871))
```

> + Clang-format pointer alignment change to Left ([#872](https://github.com/ginkgo-project/ginkgo/pull/872))

Reviewer 3: ```suggestion
+ Change clang-format pointer alignment to left ([#872](https://github.com/ginkgo-project/ginkgo/pull/872))
```
